brooklyn-tosca-transformer/src/test/java/io/cloudsoft/tosca/a4c/brooklyn/ToscaPlanToSpecTransformerTest.java

```diff
@@ -313,14 +324,59 @@ public void testDeploymentArtifacts() {
         assertNotNull(app);
         assertEquals(app.getChildren().size(), 1);
 
-        EntitySpec<?> tomcatServer = ToscaPlanToSpecTransformer
-                .findChildEntitySpecByPlanId(app, "tomcat_server");
+        EntitySpec<TomcatServer> tomcatServer =
+                (EntitySpec<TomcatServer>) ToscaPlanToSpecTransformer
+                        .findChildEntitySpecByPlanId(app, "tomcat_server");
         assertEquals(tomcatServer.getConfig().get(TomcatServer.ROOT_WAR),
                 "http://search.maven.org/remotecontent?filepath=io/brooklyn/example/" +
                         "brooklyn-example-hello-world-sql-webapp/0.6.0/" +
                         "brooklyn-example-hello-world-sql-webapp-0.6.0.war");
     }
 
+    @Test
+    public void testOverwriteInterfaceOnMysqlTopology() throws Exception {
+        try {
+
+            platform.uploadSingleYaml(new ResourceUtils(platform).getResourceFromUrl("brooklyn-resources.yaml"), "brooklyn-resources");
+            platform.loadTypesFromUrl(AlienSamplesLiveTest.ALIEN_SAMPLE_TYPES_GITHUB_URL, true);
+
+            String templateUrl =
+                    getClasspathUrlForResource(TEMPLATES_FOLDER + "mysql-topology-overwrited-interface.tosca.yaml");
+
+            EntitySpec<?> spec = transformer.createApplicationSpec(
+                    new ResourceUtils(mgmt).getResourceAsString(templateUrl));
+
+            // Check the basic structure
+            assertNotNull(spec, "spec");
+            assertEquals(spec.getType(), BasicApplication.class);
+
+            assertEquals(spec.getChildren().size(), 1, "Expected exactly one child of root application");
+            EntitySpec<?> compute = Iterators.getOnlyElement(spec.getChildren().iterator());
+            assertEquals(compute.getType(), BasicApplication.class);
+
+            assertEquals(compute.getChildren().size(), 1, "Expected exactly one child of root application");
+            EntitySpec<?> mysql = Iterators.getOnlyElement(compute.getChildren().iterator());
+            assertEquals(mysql.getType(), VanillaSoftwareProcess.class);
+
+            // Check the config has been set
+            assertEquals(mysql.getConfig().get(ConfigKeys.newStringConfigKey("port")), "3306");
+            assertEquals(mysql.getConfig().get(ConfigKeys.newStringConfigKey("db_user")), "martin");
+
+            // Check that the inputs have been set as exports on the scripts
+            assertFalse(mysql.getConfig().get(VanillaSoftwareProcess.LAUNCH_COMMAND).toString().contains("export PORT=3361"));
+            assertFalse(mysql.getConfig().get(VanillaSoftwareProcess.LAUNCH_COMMAND).toString().contains("export DB_USER=martin"));
+            assertFalse(mysql.getConfig().get(VanillaSoftwareProcess.LAUNCH_COMMAND).toString().contains("export DB_NAME=wordpress"));
+            assertTrue(mysql.getConfig().get(VanillaSoftwareProcess.LAUNCH_COMMAND).toString().contains("#launch mysql service"));
```

Comment:
I take it this is the key assertion that indicates the script was overwritten. It would be better to make it much more explicit - make `mock/start.sh` and this assertion say something like "OVERWRITTEN VALUE" so it's immediately apparent to the reader.